### DISCUSSION TITLE
Linking Users to Contacts

### DISCUSSION THREAD

**Discussion by JFQueralt** [cortezaproject/corteza-server#807] [Ideas]
Are there any plans to link Users with their potential Contact doppelgänger?

Motivation: I am a user of Corteza and I therefore need to have permissions applied to me as I interact with the platform.
However, I am also a Contact (by the mere fact that I belong to one of more organizations, working groups or whatever.
Instead of duplicating all my data in 2 models, it makes sense that Account holds my credentials & basic info and Contacts holds all the rest.

Also, I see the potential to allow people who have their data in my instance to access THEIR Contact data page and be able to rectify and/or delete their data. An "easy" way to look at this is by assigning "Owner" profile to the Contact itself.
Authentication could be make through a password initialization via the assigned email (or any other channel if existing - a priority sequence can be established).

**Comment by skamensky** [comment]
Would creating a lookup from the contact to the user solve this issue?
You'd still have to copy data from one to another if you want it to be displayed on a page, but you could automate that using workflows.

Also, you could use contextual roles to give people access to their contact.

**Comment by JFQueralt** [comment]
Yes, it crossed my mind. Still considering if that would be the right approach.

I'll have a look at contextual roles. I don't think I have seen anything on that.

**Reply by skamensky** [reply]
Here's a link to help start you out

https://docs.cortezaproject.org/corteza-docs/2022.9/integrator-guide/security-model/references/contextual-role-parameters.html

**Comment by JFQueralt** [comment]
Lovely.
I did some search and found info to dive into it.
Will check your link too.
